### PR TITLE
translation: binary search updates

### PR DESCRIPTION
**Pull request by thomasq0**
If this pull request (PR) pertains to **Chinese-to-English translation**, please confirm that you have read the contribution guidelines and complete the checklist below:

- [ ] This PR represents the translation of a single, complete document, or contains only bug fixes.
- [ ] The translation accurately conveys the original meaning and intent of the Chinese version. If deviations exist, I have provided explanatory comments to clarify the reasons.

If this pull request (PR) is associated with **coding or code transpilation**, please attach the relevant console outputs to the PR and complete the following checklist:

- [ ] I have thoroughly reviewed the code, focusing on its formatting, comments, indentation, and file headers.
- [ ] I have confirmed that the code execution outputs are consistent with those produced by the reference code (Python or Java).
- [ ] The code is designed to be compatible on standard operating systems, including Windows, macOS, and Ubuntu.

### CODE REVIEW

File: en/docs/chapter_searching/binary_search.md [hunk 1]
```diff
@@ -1,24 +1,24 @@
 # Binary search
 
-<u>Binary search</u> is an efficient search algorithm based on the divide-and-conquer strategy. It utilizes the orderliness of data, reducing the search range by half each round until the target element is found or the search interval is empty.
+<u>Binary search</u> is an efficient search algorithm based on the divide-and-conquer strategy. By taking advantage of the orderliness of elements inside the sorted array, it reduces the search interval by half each round until the target element is found or the search interval is empty.
```
Comment (ofou) [inline en/docs/chapter_searching/binary_search.md:3]: <u>Binary search</u> is an efficient search algorithm that uses a divide-and-conquer strategy. It takes advantage of the sorted order of elements in an array by reducing the search interval by half in each iteration, continuing until either the target element is found or the search interval becomes empty.

**Review by ofou — CHANGES_REQUESTED** [review]

File: en/docs/chapter_searching/binary_search.md [hunk 2]
```diff
@@ -1,24 +1,24 @@
...
 
 !!! question
 
-    Given an array `nums` of length $n$, with elements arranged in ascending order and non-repeating. Please find and return the index of element `target` in this array. If the array does not contain the element, return $-1$. An example is shown in the figure below.
+    Given an array `nums` of length $n$, where elements arranged in ascending order without duplicates. Please find and return the index of element `target` in this array. If the array does not contain the element, return $-1$. An example is shown in the figure below.
```
Comment (SamJin98) [inline en/docs/chapter_searching/binary_search.md:7]: 'where elements arranged in ascending order without duplicates.'
to
'where elements are arranged in ascending order and are without duplicates.'
for better readability and precision.

File: en/docs/chapter_searching/binary_search.md [hunk 3]
```diff
@@ -41,43 +41,43 @@ If the array does not contain the target element, the search interval will event
...
+- Binary search can only be applied to sorted data. Unsorted data must be sorted before applying binary search, which may not be worthwhile as sorting algorithm typically has a time complexity of $O(n \log n)$. Such cost is even higher than linear search, not to mention binary search itself. For scenarios with frequent insertion, the cost of remaining the array in order is pretty high as the time complexity of inserting new elements into specific positions is $O(n)$.
+- Binary search may use array only. Binary search requires non-continuous (jumping) element access, which is inefficient in linked list. As a result, linked list or data structures based on linked list may not be suitable for this algorithm.
+- Linear search performs better in small dataset. In linear search, only 1 decision operation is required for each iteration; whereas in binary search, it involves 1 addition, 1 division, 1 to 3 decision operations, 1 addition (subtraction), totaling 4 to 6 operations. Therefore, if data size $n$ is small, linear search is faster than binary search.
```
Comment (SamJin98) [inline en/docs/chapter_searching/binary_search.md:83]: 'in small dataset'
to
'on small dataset'

**Review by SamJin98 — COMMENTED** [review]
some minor grammar error :)

File: en/docs/chapter_searching/binary_search.md (hunk 1 above)
Comment (thomasq0) [reply en/docs/chapter_searching/binary_search.md]: accepted

**Comment by thomasq0** [comment]
> some minor grammar error :)

Thanks.

**Review by krahets — APPROVED** [review]

**Pull request merged**